### PR TITLE
Skip the example with the drive letter of Windows

### PR DESCRIPTION
**Pull request by hsbt**
# Description:

In GitHub Actions, The drive letter returns a downcase value.

```
  You have changed in the Gemfile:\n* rack from `no specified source` to `D:/a/rubygems/rubygems/bundler/tmp/2/libs/rack (at master@d8e48a3)`
```
  vs
```
  You have changed in the Gemfile:
  * rack from `no specified source` to `d:/a/rubygems/rubygems/bundler/tmp/2/libs/rack (at master@d8e48a3)`
```

## What was the end-user or developer problem that led to this PR?

The examples of Windows environment is always failed in the current master branch.

## What is your fix for the problem, implemented in this PR?

Skip the failing exapmles in Windows environment.
______________

# Tasks:

- [x] Describe the problem / feature
- [ ] Write tests
- [x] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
I'm curious about how this started failing. Was this a change in the github actions environment or a change we made?

**Comment by MSP-Greg** [comment]
I believe that Actions Windows images have always had the HDD as `C:` (uppercase) and the SSD drive as `d:` (lowercase).  The repo being tested is always on the SSD (`d:`). Since Actions flushes/removes older logs, it's impossible to verify.

Hence, I believe this is a RG/Bundler change.

**Comment by deivid-rodriguez** [comment]
Thanks, that's useful. It should be a matter of bisecting the latest commits.

**Comment by MSP-Greg** [comment]
JFYI,

Currently, Actions Windows does not have an `ENV['TMPDIR']` setting.  Ruby then uses another ENV setting (`TEMP` or `TMP`) for the path of the tempdir.  On Actions, and most Windows installs, those point a temp folder on the same drive as the OS, hence, `C:`.

For a long time, it has been set in [MSP-Greg/setup-ruby-pkgs](https://github.com/MSP-Greg/setup-ruby-pkgs) to the value of `ENV['RUNNER_TEMP']`, which is on the SSD (`d:`).  Lot's of extension gems are using the Action for updating gcc and installing packages (cross-platform).

I believe the systems currently used for Actions CI are eight core systems, with VM's providing two cores to each CI job.  Depending on lots of factors, the HDD may slow down IO, so that was why `setup-ruby-pkgs`  changed the temp folder to `d:`.

Long story short, `ruby/setup-ruby` may also change this setting in the near future.  I wanted to mention this, but it shouldn't affect this issue...

**Comment by MSP-Greg** [comment]
@deivid-rodriguez 

Well, I just added some checks to my 'Actions runner setup/configuration' repo.

PowerShell has a variable `$PSScriptRoot`, which is equivalent to `__dir__`.  I ran both types of scripts, and both show the SSD as `D:`.

Also, the ENV variables pointed to the SSD have a been changed to `D:`.  The last time I ran it (2020-07-13), all the ENV variables used a lowercase `d:`.  Sorry about the previous info.

**Comment by deivid-rodriguez** [comment]
Ok, so change in Github Actions after all. In order to make our specs stronger, should we normalize casing of the affected environment variables?
